### PR TITLE
Aden-4814 TNG Top leaderBoard Test

### PR DESCRIPTION
**Pull request by bulaszenko**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VuapVideos.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  public static final String PORVATA_VAST = "porvata_vast";
+
+  public static final String VAST_VIDEO = "', '<?xml version=\"1.0\" encoding=\"UTF-8\"?><VAST xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance" +
+      "\" xsi:noNamespaceSchemaLocation=\"vast_2.0.1-creativeView.xsd\" version=\"2.0\"><Ad id=\"wikia\"><InLine>" +
+      "<AdSystem>Wikia</AdSystem><AdTitle>Wikia Red Test Video</AdTitle><Impression></Impression><Creatives>" +
+      "<Creative><Linear><Duration>00:00:03</Duration><MediaFiles><MediaFile type=\"video/webm\" height=\"360\" width=\"640\" delivery=\"progressive\">" +
+      "<![CDATA[http://public.adeng.wikia-dev.pl/porvata/red.webm]]></MediaFile></MediaFiles></Linear></Creative></Creatives></InLine></Ad></VAST>'";
+
+  public static final String VAST_VIDEO_WITH_TRACKING_EVENTS = "', '<?xml version=\"1.0\" encoding=\"UTF-8\"?><VAST xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance" +
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VuapVideos.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VuapVideos.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package com.wikia.webdriver.pageobjectsfactory.componentobject.ad;
+
+public class VuapVideos {
+
+  private VuapVideos() {
+    throw new IllegalAccessError("Utility class");
+  }
+
+  public static final String PORVATA_VAST = "porvata_vast";
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VuapVideos.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VuapVideos.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  public static final String VAST_VIDEO = "', '<?xml version=\"1.0\" encoding=\"UTF-8\"?><VAST xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance" +
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VuapVideos.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+public class TestAdsVuapTheNewGeneration extends TemplateNoFirstLoad {
+
+  private static final long MAX_AUTOPLAY_MOVIE_DURATION = 40L;
+
+  private static final String FANDOM_ARTICLE_WESTWORLD_LINK = "http://adeng.fandom.wikia.com/articles/whats-coming-westworld-finale";
+
+  private static final String PROJECT43_TNG_ARTICLE_LINK = "http://project43.wikia.com/wiki/DevTemplates/VUAP/TNG";
+
+  private AdsBaseObject openPageWithVideoInLocalStorage(Page page, String player, String video) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java]: MAJOR Remove this unused private "openPageWithVideoInLocalStorage" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* MAJOR 2 major
* MINOR 4 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AutoplayVuap.java#L204](https://github.com/Wikia/selenium-tests/blob/a0fbfc41111cf805ac09adb73d33aa155e0e41e6/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java#L204): Remove this unused private "isOverlayVisible" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MINOR [AdsDataProvider.java#L1331](https://github.com/Wikia/selenium-tests/blob/a0fbfc41111cf805ac09adb73d33aa155e0e41e6/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1331): Define a constant instead of duplicating this literal " .video-player iframe" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java [hunk 5]
```diff
@@ -14,21 +14,39 @@
...
+
+  private static final String REPLAY_CLASS_NAME = "replay-overlay";
 
   private static final String CURRENT_TIME_CLASS_NAME = "current-time";
 
   private static final String SPEAKER_CLASS_NAME = "speaker";
 
+  private static final String CLOSE_BUTTON_CLASS_NAME = "close-ad";
+
+  private static final String AD_TNG_CLICK_AREA_2_SELECTOR = "div#area2";
```
Comment (mleczey) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: `div` is not needed, if you're using id `#area2`

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java [hunk 6]
```diff
@@ -14,21 +14,39 @@
...
+
+  private static final String AD_TNG_CLICK_AREA_4_SELECTOR = "div#area4";
```
Comment (mleczey) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: same here

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java [hunk 7]
```diff
@@ -14,21 +14,39 @@
...
+
+  private static final String AD_RESOLVED_STATE_IMAGE_SELECTOR = "#background2";
```
Comment (mleczey) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: and here

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java [hunk 8]
```diff
@@ -81,17 +99,39 @@ public void play() {
     }
   }
 
-  public void clickOnImage() {
-    final WebElement iframe = driver.findElement(By.cssSelector("#" + slot + " .provider-container iframe"));
-    driver.switchTo().frame(iframe);
-    driver.findElement(By.id("background_right")).click();
-    driver.switchTo().defaultContent();
+  public void replay() {
+      clickElement(String.format(REPLAY_BUTTON_SELECTOR_FORMAT, slot));
```
Comment (mleczey) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: To be consistent I would add `if` check if video is really not playing.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java [hunk 9]
```diff
@@ -81,17 +99,39 @@ public void play() {
...
+      playing = true;
+  }
+
+  public void clickOnDefoultStateAdImage() {
```
Comment (mleczey) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: typo `Defoult`

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java [hunk 10]
```diff
@@ -81,17 +99,39 @@ public void play() {
...
+    clickOnAdImage(AD_DEFAULT_STATE_IMAGE_SELECTOR);
+  }
+
+  public void clickOnDefaultAdTngClickArea2() {
```
Comment (mleczey) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: I think naming it `clickOnClickArea2` would be enough. What do you think?

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java [hunk 11]
```diff
@@ -81,17 +99,39 @@ public void play() {
...
+    clickOnAdImage(AD_TNG_CLICK_AREA_2_SELECTOR);
+  }
+
+  public void clickOnDefaultAdTngClickArea4() {
```
Comment (mleczey) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: Same here

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java [hunk 12]
```diff
@@ -102,6 +142,7 @@ public double getCurrentTime() {
 
   public double getVideoHieght() {
     if (playing) {
+      pause();
```
Comment (mleczey) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: Now this method name is not really describing what it is doing (pausing + getting video height)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+    runScript.execute("localStorage.setItem('" + player + video + ");");
+    ads.getUrl(page);
+    return ads;
+  }
+
+  @Test(groups = "AdsVuapDefaultStateAutoplayOasis",
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsVuapTngDesctop")
+  public void vuapDefaultStateShouldStartPlayingAdvertisementAutomatically(Page page, String slot, String videoIframeSelector) {
+    openPageWithVideoInLocalStorage(page, VuapVideos.PORVATA_VAST, VuapVideos.VAST_VIDEO);
```
Comment (mleczey) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java]: I see that you pass those parameters a lot `VuapVideos.PORVATA_VAST, VuapVideos.VAST_VIDEO`. Maybe you could extract method from it or hardcode those two parameters?

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+  @Test(groups = "AdsVuapDefaultStateMuteOasis",
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsVuapTngDesctop")
+  public void vuapDefaultStateAutoplayIsMutredAfrerReplayIsNotMuted(Page page, String slot, String videoIframeSelector) {
+    openPageWithVideoInLocalStorage(page, VuapVideos.PORVATA_VAST, VuapVideos.VAST_VIDEO);
+    final AutoplayVuap vuap = new AutoplayVuap(driver, slot, videoIframeSelector);
+
+    VuapAssertions.verifyVideoUnmuteAndMute(vuap);
+
+    VuapAssertions.verifyVideoEndedAndReplyButtonDisplayed(vuap, MAX_AUTOPLAY_MOVIE_DURATION);
```
Comment (mleczey) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java]: Maybe you could split this assertion int two?

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+    final AdsBaseObject ads = openPageWithVideoInLocalStorage(page, VuapVideos.PORVATA_VAST, VuapVideos.VAST_VIDEO);
+    final AutoplayVuap vuap = new AutoplayVuap(driver, slot, videoIframeSelector);
+
+    ads.refreshPage();
+
+    vuap.replay();
+
+    vuap.pause();
+
+    vuap.clickOnCloseButton();
```
Comment (mleczey) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java]: What do you think about renaming this method to simple `close`?

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+      dataProvider = "adsVuapTngDesctop")
+  public void vuapResolvedStateIsNotMute(Page page, String slot, String videoIframeSelector) {
+    AdsBaseObject ads = openPageWithVideoInLocalStorage(page, VuapVideos.PORVATA_VAST, VuapVideos.VAST_VIDEO);
+    final AutoplayVuap vuap = new AutoplayVuap(driver, slot, videoIframeSelector);
+
+    ads.refreshPage();
+
+    vuap.replay();
+
+    vuap.clickOnSoundButton();
```
Comment (mleczey) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java]: Are muting/unmuting video here?

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java (hunk 8 above)
Comment (bulaszenko) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: discussed offline it stays like it is

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AutoplayVuap.java#L200](https://github.com/Wikia/selenium-tests/blob/e5fbc23bf4aa342ac6c13f0c48ab26da6a684b33/src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java#L200): Remove this unused private "isOverlayVisible" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MINOR [AdsDataProvider.java#L1331](https://github.com/Wikia/selenium-tests/blob/e5fbc23bf4aa342ac6c13f0c48ab26da6a684b33/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1331): Define a constant instead of duplicating this literal " .video-player iframe" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

**Pull request merged**